src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -200,6 +200,7 @@ public override async Task InitializeAsync()
             connectionParameters.IsRelay = this.connectionManager.ConnectionSettings.RelayTxes;
 
             connectionParameters.TemplateBehaviors.Add(new PingPongBehavior());
+            connectionParameters.TemplateBehaviors.Add(new EnforcePeerVersionCheckBehavior(this.chainIndexer, this.nodeSettings, this.network, this.loggerFactory));
```

Comment:
Why adding this for all types of nodes? We don't need this for bitcoin or stratis networks, just for cirrus.

Reply:
We don't need this for other networks but at the same time without explicit configuration in the network class this functionality will not be used for other networks.

Reply:
Still this contaminates code that is not using it, can you not just add this in the POA behaviour which is more correct.